easybuild/tools/github.py

```diff
@@ -718,12 +761,181 @@ def update_pr(pr, paths, commit_msg=None):
     print_msg(msg, log=_log, prefix=False)
 
 
+def check_github():
+    """
+    Check status of GitHub integration, and report back.
+    * check whether GitHub username is available
+    * check whether a GitHub token is available, and whether it works
+    * check whether git and GitPython are available
+    * check whether push access to own GitHub repositories works
+    * check whether creating gists works
+    * check whether location to local working directories for Git repositories is available (not strictly needed)
+    """
+    # start by assuming that everything works, individual checks will disable action that won't work
+    status = {}
+    for action in ['--from-pr', '--new-pr', '--review-pr', '--upload-test-report', '--update-pr']:
+        status[action] = True
+
+    print "\nChecking status of GitHub integration...\n"
+
+    # check whether we're online; if not, half of the checks are going to fail...
+    try:
+        urllib2.urlopen(GITHUB_URL, timeout=5)
+    except urllib2.URLError as err:
+        sys.stderr.write("\nERROR: checking status of GitHub integration must be done online.\n\n")
+        sys.exit(1)
+
+    # GitHub user
+    print "* GitHub user...",
+    github_user = build_option('github_user')
+    if github_user is None:
+        check_res = "(none available) => FAIL"
+        status['--new-pr'] = status['--update-pr'] = status['--upload-test-report'] = False
+    else:
+        check_res = "%s => OK" % github_user
+
+    print check_res
+
+    # check GitHub token
+    print "* GitHub token...",
+    github_token = fetch_github_token(github_user)
+    if github_token is None:
+        check_res = "(no token found) => FAIL"
+    else:
+        # don't print full token, should be kept secret!
```

Comment:
Fully printing it is a bad idea, since it's basically a password.

There's no harm done in a partial print though, since this info is not enough to steal/copy the token.

It can still be useful for debugging issues though, so I'm in favour of keeping this in.